iwyu.cc

```diff
@@ -3623,7 +3630,11 @@ class IwyuAstConsumer
   // Called whenever a variable, function, enum, etc is used.
   bool VisitDeclRefExpr(clang::DeclRefExpr* expr) {
     if (CanIgnoreCurrentASTNode())  return true;
-    ReportDeclUse(CurrentLoc(), expr->getDecl());
+    if (const UsingShadowDecl* found_decl = DynCastFrom(expr->getFoundDecl())) {
+      ReportDeclUse(CurrentLoc(), found_decl->getTargetDecl());
```

Comment:
Also, I'd love it if you could add a comment here for each of the two reports with the kind of constructs they're intended to catch.